opendrift/models/radionuclides.py

```diff
@@ -452,23 +551,37 @@ def update_speciation(self):
         # Set the new speciation
         self.elements.specie=specie_out
         
-        logging.debug('old species: %s' % specie_in[phaseshift])
-        logging.debug('new species: %s' % specie_out[phaseshift])
+        self.logger.debug('old species: %s' % specie_in[phaseshift])
+        self.logger.debug('new species: %s' % specie_out[phaseshift])
+
+
+        for iin in xrange(self.nspecies):
```

Comment:
For Python3, ```xrange``` does not exist, and one has to use ```range``` instead. It is recommended to start using Python3, as Python2 will not be supported much longer.